misc/config_tools/launch_config/launch_script_template.sh

```diff
@@ -0,0 +1,166 @@
+                    online=`cat ${cpu_path}/online`
+                done
+                echo ${processor_id} > /sys/devices/virtual/misc/acrn_hsm/remove_cpu
+            fi
+        fi
+    done
+}
+
+function unbind_device() {
+    physical_bdf=$1
+
+    vendor_id=$(cat /sys/bus/pci/devices/${physical_bdf}/vendor)
+    device_id=$(cat /sys/bus/pci/devices/${physical_bdf}/device)
+
+    echo $(printf "%04x %04x" ${vendor_id} ${device_id}) > /sys/bus/pci/drivers/pci-stub/new_id
+    echo ${physical_bdf} > /sys/bus/pci/devices/${physical_bdf}/driver/unbind
+    echo ${physical_bdf} > /sys/bus/pci/drivers/pci-stub/bind
+}
+
+function create_tap() {
+    # create a unique tap device for each VM
+    tap=$1
+    tap_exist=$(ip a | grep "$tap" | awk '{print $1}')
+    if [ "$tap_exist"x != "x" ]; then
+        echo "tap device existed, reuse $tap"
+    else
+        ip tuntap add dev $tap mode tap
+    fi
+
+    # if acrn-br0 exists, add VM's unique tap device under it
+    br_exist=$(ip a | grep acrn-br0 | awk '{print $1}')
+    if [ "$br_exist"x != "x" -a "$tap_exist"x = "x" ]; then
+        echo "acrn-br0 bridge aleady exists, adding new tap device to it..."
+        ip link set "$tap" master acrn-br0
+        ip link set dev "$tap" down
+        ip link set dev "$tap" up
+    fi
+}
+
+function mount_partition() {
+    partition=$1
+
+    tmpdir=`mktemp -d`
+    mount ${partition} ${tmpdir}
+    echo ${tmpdir}
+}
+
+function unmount_partition() {
+    tmpdir=$1
+
+    umount ${tmpdir}
+    rmdir ${tmpdir}
+}
+
+# Generators of device model parameters
+
+function add_cpus() {
+    # Each parameter of this function is considered the processor ID (as is reported in /proc/cpuinfo) of a CPU assigned
+    # to a post-launched RTVM.
+
+    if [ "${rtos_type}" != "no" ]; then
+        offline_cpus $*
+    fi
+
+    cpu_list=$(local IFS=, ; echo "$*")
+    echo -n "--cpu_affinity ${cpu_list}"
+}
+
+function add_rtvm_options() {
+    if [ "${rtos_type}" = "Soft RT" ]; then
+        echo -n "--rtvm"
+    elif [ "${rtos_type}" = "Hard RT" ]; then
+        echo -n "--rtvm"
+        echo -n "--lapic_pt"
+    fi
+}
+
+function add_interrupt_storm_monitor() {
+    threshold_per_sec=$1
+    probe_period_in_sec=$2
+    inject_delay_in_ms=$3
+    delay_duration_in_ms=$4
+
+    echo -n "--intr_monitor ${threshold_per_sec},${probe_period_in_sec},${inject_delay_in_ms},${delay_duration_in_ms}"
+}
+
+function add_logger_settings() {
+    loggers=()
+
+    for conf in $*; do
+        logger=${conf%=*}
+        level=${conf#*=}
+        loggers+=("${logger},level=${level}")
+    done
+
+    cmd_param=$(local IFS=';' ; echo "${loggers[*]}")
+    echo -n "--logger_setting ${cmd_param}"
+}
+
+function add_virtual_device() {
+    slot=$1
+    kind=$2
+    options=$3
+
+    if [ "${kind}" = "virtio-net" ]; then
+        # Create the tap device
+        tap_conf=${options%,*}
+        create_tap "tap_${tap_conf#tap=}" >> /dev/stderr
```

Comment:
Is this prefixing `tap_` to a TAP device name given by the user?

Reply:
Yes, today the virtio-net parameters are always in the form `tap=xxx,mac_seed=yyy` (AFAIK), so picking the first option before the first comma and strip the `tap=` string evaluates to exactly the tap name specified by the user.

Reply:
That's understood but my concern is that the TAP name given to the `create_tap` function adds the `tap_` prefix to the name given by the user, or am I reading this wrong?

Reply:
This is the case and is the logic we have today, which is also preserved during this rewriting. I'm not sure to what extent our validation depends on this fact, though.

Reply:
This was originally introduced by PR #4277 to resolve issue #4255. I would recommend to exclude this change from this PR and create a separate thread (perhaps another GitHub issue) to discuss with the original authors why this is needed and whether we can get rid of them.

Reply:
OK, let's do it this way.

I'm pretty sure that bug was precisely because the code was looking the `tap` keyword in the device name, something that is no longer the case. But let's validate this with the code owner and fix this in a subsequent PR.